packages/backend/src/persistance/otp/setOtp.ts

```diff
@@ -0,0 +1,44 @@
+import { stringify } from '@dm3-org/dm3-lib-shared';
+import { Redis, RedisPrefix } from '../getDatabase';
+import { getIdEnsName } from '../getIdEnsName';
+import { IOtp, NotificationChannelType } from '@dm3-org/dm3-lib-delivery';
+
+export function setOtp(redis: Redis) {
+    return async (
+        ensName: string,
+        otp: string,
+        channelType: NotificationChannelType,
+        generatedAt: Date,
+    ) => {
+        // Get existing OTP records from Redis
+        const existingOtp = await redis.get(
```

Comment:
I wonder if it wouldn't be better to have the OTP for a certain channel scoped in within the key. 
So for example when creating an OTP for Email the key would be RedisPrefix.Otp + (await getIdEnsName(redis)(ensName)) =EMAIL

That would lead to the system only have a valid OTP over time and reduce the complexity of both setting and retrieving the OTP from the DB